From d202b7143673a84998521c751770d6b4bdd01259 Mon Sep 17 00:00:00 2001
From: Zdenek Dohnal <zdohnal@redhat.com>
Date: Mon, 27 Jan 2025 10:38:38 +0100
Subject: [PATCH] xsane-save.c: Specify function argument in prototype

xsane fails to build with GCC 15 and its default -std=c23, because the
new standard has more strict data type checks. In this case it takes
empty brackets as the function does not have arguments, so if the
function later uses argument(s), it reports error.
---
 src/xsane-save.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/xsane-save.h b/src/xsane-save.h
index 890484f..1263d8f 100644
--- a/src/xsane-save.h
+++ b/src/xsane-save.h
@@ -50,7 +50,7 @@ struct pdf_xref
 
 
 extern int xsane_create_secure_file(const char *filename);
-extern void xsane_cancel_save();
+extern void xsane_cancel_save(int *cancel_save);
 extern void xsane_convert_text_to_filename(char **filename);
 extern size_t xsane_get_filesize(const char *filename);
 extern void xsane_ensure_counter_in_filename(char **filename, int counter_len);
-- 
GitLab

